pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -66,9 +66,14 @@ func ApplyMonitoringConsole(client splcommon.ControllerClient, cr splcommon.Meta
 
 	//by default assume we are adding new instances in the monitoring console configMap
 	addNewURLs := true
-
+	namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: GetSplunkStatefulsetName(SplunkMonitoringConsole, cr.GetNamespace())}
 	if cr.GetObjectMeta().GetDeletionTimestamp() != nil {
 		addNewURLs = false
+		//explicit removal of owners for MC
+		_, err = splctrl.RemoveStatefulSetOwnerRef(client, cr, namespacedName)
+		if err != nil {
+			return err
+		}
```

Comment:
Just thinking this, Assuming this change is to have the MC running even if the last CR is deleted(?), then we can avoid setting the owner reference in the first place unless the CRs are really interested in the MC STS state changes(?). If so, I guess, we don't need any of these changes.